services/streammanager/kafka/kafkamanager.go

```diff
@@ -354,78 +369,72 @@ func Produce(jsonData json.RawMessage, producer interface{}, destConfig interfac
 		return makeErrorResponse(err) //returning 500 for retrying, in case of bad configuration
 	}
 
-	topic := conf.Topic
+	if conf.Topic == "" {
+		return makeErrorResponse(fmt.Errorf("invalid destination configuration: no topic"))
+	}
 
 	if kafkaBatchingEnabled {
-		return sendBatchedMessage(jsonData, kafkaProducer, topic)
+		return sendBatchedMessage(jsonData, p, conf.Topic)
 	}
 
-	return sendMessage(jsonData, kafkaProducer, topic)
+	return sendMessage(jsonData, p, conf.Topic)
 }
 
-func sendBatchedMessage(
-	jsonData json.RawMessage, kafkaProducer sarama.SyncProducer, topic string,
-) (int, string, string) {
-	timestamp := time.Now()
+func sendBatchedMessage(jsonData json.RawMessage, p producer, topic string) (int, string, string) {
 	var batch []map[string]interface{}
 	err := json.Unmarshal(jsonData, &batch)
 	if err != nil {
-		return 400, "Failure", "Error while unmarshalling json data :: " + err.Error()
+		return 400, "Failure", "Error while unmarshalling json data: " + err.Error()
 	}
 
-	batchedMessage, err := prepareBatchedMessage(topic, batch, timestamp)
+	timestamp := time.Now()
+	batchOfMessages, err := prepareBatchOfMessages(topic, batch, timestamp)
 	if err != nil {
-		return 400, "Failure", "Error while preparing batched message :: " + err.Error()
+		return 400, "Failure", "Error while preparing batched message: " + err.Error()
 	}
 
-	err = kafkaProducer.SendMessages(batchedMessage)
+	err = p.Publish(context.TODO(), batchOfMessages...)
```

Comment:
shouldn't we use here a timeout context using as timeout the `Opt.Timeout` passed in the producer function(s)?